mllib/src/main/scala/org/apache/spark/ml/optim/WeightedLeastSquares.scala

```diff
@@ -439,8 +439,9 @@ private[ml] object WeightedLeastSquares {
 
     /**
      * Weighted population standard deviation of labels.
+     * We prevent variance from negative value caused by numerical error.
```

Comment:
I'm not so against this, but this is really an implementation detail and not relevant to the caller. It's a value that is by definition nonnegative.